### PR TITLE
feat(stellar): improve Stellar wallet UX

### PR DESCRIPTION
**Pull request by akbarsaputrait**
## Summary

Improve Stellar wallet UX, add Direct Settlement support, and fix a set of payment-race and bundle-size issues found while testing this flow. Approach: keep Stellar wallet selection always visible (extension or WalletConnect fallback), auto-route same-chain Stellar payments through a direct settlement path, and close several races where a stale order or a duplicated init could send a payment down the wrong path.

### Stellar Wallet UX
- Always show Freighter & LOBSTR in the wallet list, even without the browser extension; route through WalletConnect when the extension isn't available
- Detect WalletConnect QR modal close and reject properly instead of leaving a stuck spinner
- Pre-fetch wallets in `StellarContextProvider` and cache the kit synchronously — removes the loading spinner and remount flash on the Connect Stellar page
- Dynamic description text: "Scan QR" for WalletConnect vs "Open wallet" for extension

### Stellar Direct Settlement
- Auto-detects same-chain Stellar USDC/EURC payments (source chain+token === destination chain+token) and forces `intent: "stellar_direct"`
- `intent` sent as a top-level field in `POST /payments`, separate from the display `title`
- Handles `settlementMode: "stellar_direct"` in the response: source IS destination, fee "0.00", direct send
- Added `intent` to `CreateNewPaymentParams`/`CreatePaymentRequest`, `settlementMode` to `PaymentResponse`

### Race Fixes (payId-mode source-token switching)
- `PayWithStellarToken`/`PayWithSolanaToken`: the auto-transfer effect could fire `handleTransfer` twice before either finished (Solana had no guard at all), sending duplicate checkout POSTs for the same payment. Replaced with a synchronously-claimed in-flight promise so a second call awaits the first instead of racing it.
- `usePaymentState.payWithToken` (EVM) read a closure-captured `pay.order` instead of a fresh store read, so `needRozoPayment` could evaluate against stale order data.
- `needRozoPayment` only compared `chainId`, so switching source tokens on the same chain (e.g. USDC Base → USDT ETH) skipped checkout and silently reused the stale order. Now compares chain **and** token address.
- Checkout dedup cache was keyed by `payId` alone, so switching source tokens mid-flow could return a cached result for the wrong token. Rekeyed to `payId:chainId:tokenAddress`.

### WalletConnect Double-Init
- `WalletConnectModule` was creating two independent WalletConnect Cores (one via `SignClient.init()`, one via `createAppKit()`'s internal `UniversalProvider` init) — both always fired regardless of `manualWCControl`, which only affected `close()`. Deduped to a single Core.

### Bundle Size / Perf
- `@stellar/stellar-sdk` (~14MB) was statically imported in `StellarContextProvider`, `usePaymentState`, `PayWithStellarToken`, and (via `chainAddress.ts`) `RozoPayButton` itself, forcing it to parse/execute on every mount regardless of chain used. Converted to dynamic `import()`, deferred to actual use (wallet connect, quote, tx submit). Added `validateAddressForChainAsync` for `RozoPayButton`'s internal validation path; public `isValidStellarAddress`/`isValidSolanaAddress`/`validateAddressForChain` stay sync for API compatibility.

### Other
- `RozoPayModal`/`routes.ts`: registered routes for the Stellar connect flow
- `ConnectorStellar`: use `WALLET_CONNECT_ID` constant instead of a magic string
- Workspace: `examples/nextjs-app` and `packages/connectkit` use `workspace:*` for local dev
- Expanded test coverage: `createPaymentPayload`, `derivePayIdPreferredTokens`, `isDNTEnabled`, `walletconnect.module`

## How to Test

**Wallet UX**
1. Desktop with Freighter extension installed → tapping Freighter opens the extension popup
2. Desktop without extension / incognito → tapping Freighter/LOBSTR opens WalletConnect QR
3. Freighter Mobile Explorer / mobile browser → WalletConnect modal / QR
4. Close the WalletConnect QR modal mid-connect → verify a proper rejection, no stuck spinner
5. Open the Connect Stellar page → verify no loading spinner flash

**Direct Settlement**
1. Pay Stellar USDC → Stellar USDC destination → verify `intent: "stellar_direct"` is sent and fee shows "0.00"
2. Pay Stellar EURC → Stellar EURC destination → same as above
3. Pay Stellar USDC → Stellar EURC destination (cross-token) → verify no `intent` sent, normal hub path used

**Race fixes**
1. In payId mode, rapidly switch source token twice before the first checkout resolves → verify only one checkout POST fires
2. Switch source token to a different token on the *same* chain (e.g. USDC Base → USDT Base... wait, use USDC Base → USDT ETH per commit) → verify checkout re-runs instead of reusing the stale order
3. Check console for the WalletConnect Core double-init warning — should be gone

**Perf**
1. Run `pnpm build` in `packages/connectkit`, confirm `stellar-sdk` chunk is not in the main render-critical bundle
2. Load the checkout page for a non-Stellar payment and confirm stellar-sdk isn't eagerly fetched

## Testing Checklist
- [x] Unit tests added/updated (`createPaymentPayload`, `derivePayIdPreferredTokens`, `isDNTEnabled`, `walletconnect.module`)
- [x] Manual testing completed (wallet UX matrix above)
- [ ] Integration tests for direct settlement API path
- [x] Edge cases covered (modal close, stale order, same-chain token switch)
- [x] No regressions in existing tests

## Reviewer Notes
- Bundle-size fix note from the commit itself: webpack still inlines `stellar-sdk` into the main chunk because the SDK's `index.ts` barrel re-exports the sync `isValidStellarAddress` from the same entry graph — this reduces eager execution work but doesn't yet achieve true code-splitting. A follow-up (subpath export or consumer `splitChunks` config) is still needed if full code-splitting is required.
- The three payId-mode races (duplicate checkout, stale `pay.order`, chain-only comparison) were fixed together since they all surface as variants of the same "switch source token quickly" bug — worth a closer look at `usePaymentState.ts` and both `PayWithXToken` components together rather than in isolation.

## Related
- N/A

### CODE REVIEW

File: packages/connectkit/src/components/Pages/Stellar/StellarWalletConnectQR/index.tsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        setError(err.message || "Failed to connect via WalletConnect");
+      }
+    };
+
+    connect();
+
+    return () => {
+      cancelled = true;
+    };
+  }, [stellarConnector]); // eslint-disable-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/components/Pages/Stellar/StellarWalletConnectQR/index.tsx]: Fixed. Removed dead StellarWalletConnectQR component and route — nothing was routing to it, and it had a latent signing bug (own SignClient without wcSessionPaths registration). Deleted in commit 62157552.

File: packages/connectkit/src/utils/stellar/walletconnect.module.ts [hunk 2]
```diff
@@ -155,8 +155,40 @@ export class WalletConnectModule implements ModuleInterface {
...
+      });
+      observer.observe(document.body, { childList: true, subtree: true });
+    });
+
     try {
-      const session: SessionTypes.Struct = await approval();
+      const session: SessionTypes.Struct = await Promise.race([
+        approval(),
+        modalClosed,
+      ]);
```
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/utils/stellar/walletconnect.module.ts:278]: Fixed. Captured observer outside the promise and added `finally { modalObserver?.disconnect(); }` block to ensure cleanup on both happy path and error path. Commit 62157552.

File: packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx [hunk 3]
```diff
@@ -11,74 +11,152 @@ import { SquircleIcon } from "../../../../assets/logos";
...
+        });
+      });
+
+    return options;
   }, [stellarWallets, log, setStellarConnector, setRoute]);
 
   return (
     <PageContent>
-      {isLoading || !kit ? (
+      {!kit || stellarWallets.length === 0 ? (
```
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx]: Fixed. Added `walletsLoaded` flag to StellarContextProvider — spinner only shows while `getSupportedWallets()` is in-flight, not when the result is an empty array. On error, the empty-list UI ("No Stellar wallets detected") renders instead of a perpetual spinner. Commit 62157552.

File: packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx [hunk 4]
```diff
@@ -11,74 +11,152 @@ import { SquircleIcon } from "../../../../assets/logos";
...
+              event: "click-stellar-wallet-wc",
+              walletName: wallet.name,
+            });
+          }
         },
-      }));
+      });
+    }
+
+    // Include other available wallets (excluding those already added above)
```
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx]: Fixed. Collapsed duplicate `isPlatformWrapper` and `else` branches into a single `else if (hasWalletConnect)` — both cases were identical (set WALLET_CONNECT_ID + navigate to CONNECTOR). Now only one branch handles the WalletConnect fallback. Commit 62157552.

File: packages/connectkit/src/components/Pages/Stellar/ConnectorStellar/index.tsx [hunk 5]
```diff
@@ -83,6 +83,7 @@ const ConnectorStellar: React.FC = () => {
   // Get wallet display info from stellarConnector (which is now the full wallet object)
   const walletName = stellarConnector?.name || "Stellar Wallet";
   const walletIcon = stellarConnector?.icon;
+  const isWalletConnect = stellarConnector?.id === "wallet_connect";
```
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/components/Pages/Stellar/ConnectorStellar/index.tsx]: Fixed. Now imports and uses `WALLET_CONNECT_ID` constant instead of the hardcoded `"wallet_connect"` string. Commit 62157552.

File: packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx [hunk 6]
```diff
@@ -11,74 +11,152 @@ import { SquircleIcon } from "../../../../assets/logos";
...
+              walletName: wallet.name,
+            });
+          } else {
+            // Extension NOT installed, not a platform wrapper → route through
+            // WalletConnect with AppKit modal (user picks wallet from list)
+            setStellarConnector({
+              ...wallet,
+              id: WALLET_CONNECT_ID,
+            });
+            setRoute(ROUTES.STELLAR_CONNECTOR, {
```
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx:101]: Fixed. Added `hasWalletConnect` flag to StellarContextProvider — checked during `refreshSupportedWallets()` by looking for a `wallet_connect` module in the kit's supported wallets. When the consumer provides a custom kit without WalletConnectModule, the Freighter/LOBSTR options render but clicking does nothing (no WC fallback attempted). Commit 62157552.

File: packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (akbarsaputrait) [reply packages/connectkit/src/components/Pages/Stellar/ConnectStellar/index.tsx]: Fixed. Added `walletsLoaded` flag to StellarContextProvider — set `true` in `finally` block after `getSupportedWallets()` resolves (success or error). Spinner shows only while `!walletsLoaded`, so an empty result renders the "No Stellar wallets detected" UI instead of a perpetual spinner. Commit 62157552.

**Review by shawnmuggle — DISMISSED** [review]
Reviewed the requested ecd3773..4e5dc1b range and checked the subsequent commits through current HEAD. No P0 found; approving per requested P0-only gate.

**Review by shawnmuggle — APPROVED** [review]
## Re-approval — incremental (`f7211425…7cda1ff2`) — no P0

Prior approval was dismissed by the new push. Scoped this pass to what landed since that SHA rather than re-reviewing the whole `feat/stellar-wallet-ux` diff.

### What actually changed: beta → stable pin, nothing else

Six files, and every one is the release cut:

- `packages/connectkit/package.json` — `@rozoai/intent-pay` `0.1.38-beta.16` → `0.1.38`
- `packages/pay-common/package.json` — `@rozoai/intent-common` `0.1.26-beta.5` → `0.1.26`
- `examples/nextjs-app/package.json` — both deps repinned to the stable versions
- `pnpm-lock.yaml` — specifier + resolution updates matching the above
- `CHANGELOG.md` (+57), `bundle-analysis.html` (build artifact)

**Zero source changes.** No `.ts`/`.tsx` touched in this range, so nothing in the signing or fund-routing path moved since the last review.

### Verified against the registry

Both packages are actually published, and the lockfile isn't pointing at something that doesn't exist:

```
@rozoai/intent-pay    dist-tags → { latest: '0.1.38', beta: '0.1.38-beta.16' }
```

Integrity hashes in `pnpm-lock.yaml` match the published tarballs byte for byte:

- `intent-pay@0.1.38` → `sha512-sJ6ALwejToIFOHyfXqYkOAIVvgbhk9HHqmoeZ69+BS6ChFWnYV2hCADvCv1FOjpVLTQsGzOQX6OJFtJqFKYc3w==` ✅
- `intent-common@0.1.26` → `sha512-vnIGZK8KnFmiyACV856+Ac8yVWy3IDxKGzs1hB0p73bUOY7frFrvcndQPdqXEzEFWSVkmGdtqQvYtTj0b6bGNw==` ✅

Also worth noting `latest` now resolves to `0.1.38`, so the pin and the published tag agree — no risk of consumers installing a version this repo never built.

### On the red `review` check

`review` is failing, but not on this code:

```
The template is not valid. .github/workflows/ai-pr-review-v2.yml (Line: 120, Col: 20):
Error reading JToken from JsonReader. Path '', line 0, position 0.
```

That's the AI-review workflow failing to parse its own YAML — it dies before it ever looks at the diff. Infra bug in the review tooling, not a signal about this PR. `Security scan`, `release`, and `Vercel` all pass. Worth fixing separately since it will red-flag every PR in the repo until it is.

### Verdict

Re-approving. The prior P1s (WalletConnect `getAddress()` reusing `live[0]`, etc.) are unchanged and still non-blocking — this push didn't touch them and didn't introduce anything new.
